### ISSUE TITLE
Make WinUI App Window Same Size as UWP App

### ISSUE THREAD

**Issue by AathifMahir** [microsoft/WindowsAppSDK#986]
Currently, UWP app and WinUI Desktop app has Different Initial App Window Size, It would be great if we have same size as UWP app for WinUI 3 Desktop App. Current WinUI 3 Desktop App Window is Wider but when it comes to UWP it's taller and gives more space for contents than WinUI 3 App Window in My Opinion

# Proposal: [your title here] 

Make WinUI App Window Same Size as UWP App

## Summary

App Window Size Consistency

## Rationale

* This would make Apps Consistent regardless of App Model

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish Consistency regardless of App Model | Must |
| This proposal will allow end users to have consistent experience across the apps | Must |

## Screenshots

![Screenshot 2021-06-25 182227](https://user-images.githubusercontent.com/17165402/123428213-3184a080-d5e3-11eb-842e-c3d5746b8974.png)
![Screenshot 2021-06-25 182158](https://user-images.githubusercontent.com/17165402/123428223-347f9100-d5e3-11eb-94f9-3d16c6a539ed.png)

**Comment by LanceMcCarthy** [comment]
This can be easily done with `WinUIEx` (see https://github.com/dotMorten/WinUIEx/). It can not only set the initial window's size and location, but also remember it for next time.

*I use it to fix this WinUI 3 problems in my .NET MAUI and WinUI 3 projects

```csharp
// In App.xaml.cs
var manager = WinUIEx.WindowManager.Get(window);
manager.PersistenceId = "MainWindowPersistanceId"; // this remembers the last position and size

// set minimums
manager.MinWidth = 640;
manager.MinHeight = 480;

// or just Center the window at every launch
// manager.CenterOnScreen(1024,768); 
```

**Comment by d2dyno1** [comment]
WinUIEx is a great library, however such simple features should be present out of the box for WASDK.

**Comment by LanceMcCarthy** [comment]
Yep, make sure your voice is head here (they don't follow other repos) https://portal.productboard.com/winappsdk/1-windows-app-sdk/tabs/2-planned

**Comment by riverar** [comment]
I don't think the team uses Product Board anymore. @BenJKuhn can you confirm/deny? Or maybe @dotmorten can share his experience with PB here?

**Comment by dotMorten** [comment]
@riverar According to the product board, "Windowing APIs" have already been released 🤦‍♂️

**Comment by teohumeau** [comment]
Any update on this?

**Comment by LanceMcCarthy** [comment]
@teohumeau Just use Morten's helper and avoid the headache altogether.

```csharo
using WinUIEx;
```

// To set an explicit size on launch
```csharp
const int width = 1920;
const int height = 1080;
const int x = 3440 / 2 - width / 2;
const int y = 1440 / 2 - height / 2;
window.MoveAndResize(x, y, width, height);
```

The better option is to remember the user's window size and position, the next time the app launches it will put the app window where it was the last time it was closed.
```csharp
var manager = WindowManager.Get(window);
manager.PersistenceId = "MainWindowPersistanceId";
manager.MinWidth = 640;
manager.MinHeight = 480;
```

I prefer the 2nd option, and I even use it in my .NET MAUI projects (see [this example](https://github.com/LanceMcCarthy/XkcdViewer/blob/8ec3b286b5d5e2bda29166db5d227e3aed18cb52/src/XkcdViewer.Maui/MauiProgram.cs#L69-L86)).

**Comment by ssparach** [comment]
Thanks for the feedback and discussion on this issue. WinUI and UWP are fundamentally different app models, each with distinct windowing behaviors. Because of these differences, aligning their default window sizes is not feasible as a platform-level change. If consistent window sizing is still desired, we recommend explicitly setting the app window size in your code, as suggested in earlier comments. This gives you more control over the initial window dimensions and behavior. Given this, we’re closing this issue. Please feel free to continue the discussion or open a new issue if you encounter specific technical blockers with WinUI.
